src/components/BlogPost.js

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import { ListGroupItem, Collapse } from 'reactstrap';
+import BlogContextMenu from './BlogContextMenu';
+import CommentsSection from './CommentsSection';
+import CommentEntry from './CommentEntry';
+
+class BlogPost extends React.Component {
+  constructor(props) {
+    super(props);    
+    
+    this.onPost = this.onPost.bind(this);
+    this.submitEdit = this.submitEdit.bind(this);
+
+    this.childElement = React.createRef();
```

Comment:
This isn't a react-y way of making this work - instead of having a parent call a function on the child, have the parent manage the child's state and then pass down the value + a toggle callback. I've avoided using refs anywhere in the rest of the react code, and I think that's a good constraint to keep unless there's a really good reason not to.

See [here](https://reactjs.org/docs/lifting-state-up.html)